Improve the readability of an error message

InMatrix opened the pull request:
This PR is intended to improve the readability of the error message from `debugCheckHasMaterial`. It adds a few more line breaks and section headings to make the message easier to scan.

InMatrix committed:
Combine two lines to a single "headline"

InMatrix committed:
added subheadings and line spacing

InMatrix committed:
Add additional linebreaks to separate out the "headline"

Hixie commented on packages/flutter/lib/src/material/debug.dart:
```diff
@@ -24,28 +24,29 @@ bool debugCheckHasMaterial(BuildContext context) {
         '${context.widget.runtimeType} widgets require a Material '
-        'widget ancestor.'
+        'widget ancestor, but none was found.\n'
       );
```
we should either using `writeln` or `\n` but not both. Using both is confusing.

Hixie replied:
(it's fine to use `write` and have `\n\n`)

Hixie commented on packages/flutter/lib/src/material/debug.dart:
```diff
@@ -24,28 +24,29 @@ bool debugCheckHasMaterial(BuildContext context) {
+        'widget ancestor, but none was found.\n'
       );
       message.writeln(
+        'Explanation\n'
```
since we don't have colour or font size or underlining or anything, maybe make these section headings all-uppercase?

Hixie commented on packages/flutter/lib/src/foundation/assertions.dart:
```diff
@@ -288,7 +288,7 @@ class FlutterError extends AssertionError {
         if (message.startsWith(prefix))
           message = message.substring(prefix.length);
-        debugPrint('The following $errorName was $verb:\n$message', wrapWidth: wrapWidth);
+        debugPrint('\nThe following $errorName was $verb:\n\n$message', wrapWidth: wrapWidth);
```
Is the leading one here useful? There's a horizontal line immediately above this line, so it's not something that could get confused with another paragraph.

Hixie commented:
We should definitely test this. (Looks like it's already tested, and we just need to update the relevant tests.)

InMatrix committed:
Update error reporting tests

InMatrix committed:
Merge branch 'master' of github.com:flutter/flutter into error-message

InMatrix committed:
Update assertions_text

The text now expects an empty line between the context and the exception description.

InMatrix committed:
Update material/debug_test.dart

Update debug_test.dart to reflect changes in material/debug.dart

InMatrix committed:
Update material/text_field_test

InMatrix commented:
I'm withdrawing this PR in favor of a more comprehensive solution we're considering.
